openedx/frontend-app-authoring
# feat: Use configured DEFAULT_GRADE_DESIGNATIONS

## Description

Support was recently added to edx-platform to add customized default grade designations, this change adds support for them to this MFE as well to bring it to parity with the edx-platform UI.

It also refactors the grading-settings page to use React Query and updates the logic used when partitioning grades by default to make it work better when there are more than 5 partitions.

The original logic was to subdivide each segment as half of the previous segment, so when you go from Pass/Fail to A, B, F, it will keep F at 50% and split Pass into A and B of 25% each. Adding another segment will then split B into B and C each getting 12% roughly, adding another segment would now split C into C and D of roughly 6% each.  Adding yet another segment at this point would then cause the splits of 3% and make the segment too small to be usable.  

This change will update this to switch to making even splits when you go beyond 5 segments so that all grades get equal space. Author's can then choose to readjust as needed. 

## Testing instructions

1. Change `DEFAULT_GRADE_DESIGNATIONS` in cms's settings 
2. Visit the grading setting for a course. 
3. Add segments to the grading scale. It will stop after 5 divisions A, B, C, D, F, and the `DEFAULT_GRADE_DESIGNATIONS`  will be ignored
4. Check out this PR. 
5. The segment names should now be pulled from `DEFAULT_GRADE_DESIGNATIONS`.

## Other information

[Using configured grade segments](https://github.com/user-attachments/assets/8d16673d-15c4-4a90-95d6-1e70f81e1c27)

[Better support for smaller segments](https://github.com/user-attachments/assets/fcfafb92-71f8-4431-8d4a-727c91320edf)
